lib/prometheus_exporter/server/collector.rb

```diff
@@ -38,10 +38,8 @@ def process_hash(obj)
         if collector = @collectors[obj["type"]]
           collector.collect(obj)
         else
-          metric = @metrics[obj["name"]]
-          if !metric
-            metric = register_metric_unsafe(obj)
-          end
+          metric = @metrics[obj["name"]] || register_metric_unsafe(obj) or
```

Comment:
`||` and `or` in the same line is terrifying. perhaps

```
metric = @metrics[obj["name"]] || register_metric_unsafe(obj)
if !metric
   return
end
```

I know it is way more verbose but it is less scary

Reply:
Ha, fair enough. How about I meet you half way?

```ruby
# register_metric_unsafe will log an error if it returns nil, so that code path can end here.
return unless metric
```

I can leave out the comment if you prefer, but I almost always include them with early returns/guard clauses.